strike-data-worker/src/main/java/org/ride/updater/RideUpdater.java

```
@@ -0,0 +1,73 @@
+package org.ride.updater;
+
+import org.strike.container.RideDrivers;
+import org.strike.domain.Ride;
+import org.strike.messaging.format.RideAdded;
+import org.strike.messaging.format.RideRebid;
+import org.strike.services.NotificationService;
+import org.strike.services.RideInvitationManager;
+
+/**
+ * Class implements RideUpdate listener encapsulate methods to handle any ride update (Ride added ,
+ * rebid ..)
+ * 
+ * @author mina.shafik
+ *
+ */
+public class RideUpdater implements RideUpdateListener {
+
+  RideInvitationManager rideManger;
+  NotificationService notificationService;
+
+  @Override
+  public void didReceiveRideAdded(RideAdded rideAdded) {
+
+    Ride ride = rideManger.getRide(rideAdded.getRideId());
+    // handle ride invitations
+    RideDrivers rideDrivers = rideManger.handleNewRide(ride);
+
+    // send notification for priority drivers
+    notificationService.addAvailableRide(rideDrivers.getAvailableDrivers(), rideDrivers.getRide());
+
+  }
+
+  @Override
+  public void didReceiveRideRebid(RideRebid rideRebid) {
+
+    Ride ride = rideManger.getRide(rideRebid.getRideId());
+
+    // handle ride invitations
+    RideDrivers rideDrivers = rideManger.handleRebidRide(ride, rideRebid.getOldRideId());
+
+    // send notification for priority drivers
+    notificationService.addRebidRideNotifications(rideDrivers.getRide(),
+        rideDrivers.getClaimedDrivers(), rideDrivers.getAvailableDrivers());
```

We need to apply the fail fast approach in case of errors while processing messages. Some errors should stop the worker while other shouldn't. We need to exist the worker in case of any critical errors, e.g. invalid message format from strike server, missing required fields in ampq message sent from the server,.. etc